### DISCUSSION TITLE
[Feature Request] Link generation for jumping to a function / method definition

### DISCUSSION THREAD

**Discussion by david-mcafee** [community/community#13292] [Other Feature Feedback, Questions, & Ideas]
Hi!

While writing internal documentation in a GitHub repo, it would be great if I could link to a [definition that has been generated](https://docs.github.com/en/repositories/working-with-files/using-files/navigating-code-on-github#jumping-to-the-definition-of-a-function-or-method). 

Permalinks don't fit the bill because eventually, the code linked to will be outdated. Links with line numbers also don't work, because files change and eventually those link will be outdated.

**Comment by dcreager** [comment]
Hi @david-mcafee, thanks for the feedback!  This is a great suggestion, and one that we've been looking at internally.  There are some interesting wrinkles (e.g. what if there are multiple entities that are defined with the same name?) but nothing that's a deal-breaker.

**Reply by david-mcafee** [reply]
Thanks for the response @dcreager! In the case of entities with the same name, if there were some ability to jump to a comment definition, that would be ideal (but I realize that isn't even supported ATM by definitions). But in most cases, I imagine entities with the same name wouldn't be all that common? Personally, if the definition simply went to the first defined entity, that would be fine with me.
